dags/common/rest_api_service.py

```diff
@@ -0,0 +1,35 @@
+class RestApi:
```

Comment:
I think we should review how this class is being used.

It's a detail but this class is not a RestApi but a RestApiClient, the API il already being served elsewhere.

I think the class does a good job in being generic, but it should not be used as it currently is. Here is why :
- What you are handling are "responses". I think it's better to think in business logic when you can. Our repositories are not handling responses, they're handling files.

I think this class should be used a a base (maybe with a has-a relationship instead of is-a) for the other API clients. We should have a API client for APS and one for each publisher that needs it. Each API client should know how to configure itself, and should serve functions that allows us to directly understand what we are getting from the client.

The pieces of code that will use the API client don't care about headers and parameters, they care about "give me the articles jsons from last month to today", and I think that's how we should design this service

I hope I was clear enough 😆